infra/modules/providers/azure/keyvault/main.tf

```diff
@@ -27,6 +25,16 @@ resource "azurerm_key_vault" "keyvault" {
     certificate_permissions = var.keyvault_certificate_permissions
   }
 
+  # This block configures VNET integration if a subnet whitelist is specified
+  dynamic "network_acls" {
+    for_each = length(var.subnet_id_whitelist) == 0 ? [] : [var.subnet_id_whitelist]
+    content {
+      bypass         = "None"
+      default_action = "Deny"
```

Comment:
Should this be "deny" if it's for creating whitelists?

Reply:
As I understand you need to specify 2 things in order to restrict access to a, pardon my pun, subset of subnets. (1) is to specifically deny all traffic by default and (2) is to specifically accept traffic from the specified subnets